packages/@react-native-windows/cli/src/runWindows/utils/autolink.ts

```diff
@@ -591,6 +592,155 @@ async function updateAutoLink(
   }
 }
 
+export function ensureWinUIDialect(
+  slnFile: string,
+  windowsProjectConfig: WindowsProjectConfig,
+  checkMode: boolean,
+) {
+  const buildFlagsProps = path.join(path.dirname(slnFile), 'BuildFlags.props');
+  let changesNeeded = false;
+  if (fs.existsSync(buildFlagsProps)) {
+    const buildFlagsContents = configUtils.readProjectFile(buildFlagsProps);
+    const useWinUI3FromConfig = windowsProjectConfig.useWinUI3;
```

Comment:
Do we have to support setting the flag in two places? Can we just take the value from react-native.config.js (which rather than ever being undefined, has a default if not specified) and always override what's in BuildFlags.props?

Reply:
that would mean that if you want winui3 you _have_ to have a react-native.config.js which seemed more of a divergence than I felt comfortable with, but we can discuss more if you feel strongly about it :)